### ISSUE TITLE
Composite finals should be on the same level.

### ISSUE THREAD

**Issue by JBrase** [silnrsi/script-miao#3]
AC: 
- Currently the 2nd final gets bumped upwards when in the shoulder or the waist position.
- Composite finals are not centring at all but shifted right except when the base char is U+16F00 or U+16F33.  They should be centred.

MH thought that these two are the same issue.

MH has indicated that he has dealt with this in Graphite, but not OT.

(email from AC 2015-05-06)

**Comment by mhosken** [comment]
This bug needs a language and test sequence

**Comment by JBrase** [comment]
@MH, at the risk of sounding dumb, why does it need language? Might it not apply to all languages?

**Comment by mhosken** [comment]
Yes it probably does, but given the track record of language specific rendering that Miao has, it would really help to have a test sequence and the language for all bugs. In this case, the language can probably slide.

BTW does the shift right with U+16F00 contrast with bug #5 ? I think I'm not understanding from the bug description what this bug is actually about. No image, no test string and some vaguely worded problem. I need something I can repeat.

**Comment by JBrase** [comment]
I've identified this issue in some test data:
![image](https://user-images.githubusercontent.com/7561110/31697741-1474e9d0-b37f-11e7-80ed-2dd390d35293.png)
Comparing this output with the Graphite output, in the first word, 16F6E and 16F74 have the same baseline--their positions match the Graphite output. In the fourth word, 16F74 is slightly higher than than 16F6E--its position is 175 units higher than in the Graphite output.
![image](https://user-images.githubusercontent.com/7561110/31697849-b0acd010-b37f-11e7-9ce8-cb9491c2027f.png)

**Comment by JBrase** [comment]
This problem is caused by the way vowel chains are processed, and by the fact that the raised baselines of the .T and .R glyph variants do not match the height of the S and R APs. (I suspect heights of S and R were changed at some point, but baselines of the .T and .R variants were not adjusted to match.) 

Vowel chain strategy (using 16F90 (tone top right) as an example):

1.  in sequence C + V1 + V2 ... + 16F90, insert duplicate tone between C and V1, yielding C + 16F90 + V1 + V2 ... + 16F90
2. 16F90 + V1 is then converted to V1.T. 
3. V1.T + V2 is then converted to V1.T + V2.T. 
4. V2.T + 16F90 is then converted to V2.T.
5. V1.T is attached to the S AP. (which lowers the mark by 175 units because of the difference in the heights of the APs.)
6. attempts to attach V2.T to V1.T fail because V2.T does not have the correct APs to form the chain. (.T vowels have only SEntry an VExit APs.) Consequently V2.T is simply positioned to the right of V1.T. But because V1.T was shifted downward by step 5, the heights of the two vowels will not match.

This same problem exists for the .R vowels. I don't have test data at hand, but given the examples that Nathan showed, and the comparative height of the R AP vs the baseline of the R vowels, the problem is more severe for the .R vowels than for the .T vowels.

There are several ways to fix this:

1. Adjust the heights of all of .T and .R vowels.
2. Add VEntry APs to all .T and .R vowels.
3. Make major changes to the OT so that only the first vowel of a chain is converted to its .T or .R variant. (The default vowel glyphs contain VEntry APs, and I think they should chain properly.)
